### PR TITLE
Add Api doc configuration in yaml file

### PR DESCRIPTION
**Pull request by oktapodia**
You can allow yaml or annotation format for api doc configuration

### CODE REVIEW

**Comment by oktapodia** [comment]
Following https://github.com/nelmio/NelmioApiDocBundle/pull/592

**Comment by willdurand** [comment]
I like this PR, however I'd like to merge it in a couple of weeks. I want to be sure this bundle is stable enough first (cf. https://github.com/nelmio/NelmioApiDocBundle/releases/tag/2.10.0).

**Comment by oktapodia** [comment]
Ok, don't worry, do you want some help to stabilize this bundle ?

**Comment by willdurand** [comment]
Yep, sure! I need people to test latest version in their projects, and to sort the existing issues.

**Comment by willdurand** [comment]
changed the milestone to get symfony 3.0 support first.

File: Annotation/ApiDoc.php [hunk 1]
```diff
@@ -236,6 +236,7 @@ public function __construct(array $data)
 
         if (isset($data['statusCodes'])) {
             foreach ($data['statusCodes'] as $statusCode => $description) {
+
```
Comment (OskarStark) [inline Annotation/ApiDoc.php]: why?

File: Extractor/DocumentationFilesExtractor.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+{
+    /**
+     * @var ApiDocFactory
+     */
+    protected $apiDocFactory;
+
+    /**
+     * @var array
+     */
+    protected $ddocumentationFilesConfiguration;
```
Comment (OskarStark) [inline Extractor/DocumentationFilesExtractor.php]: double D in variable name

File: Resources/doc/index.md [hunk 3]
```diff
@@ -288,6 +288,36 @@ then the `premium` view will be located at:
...
 
 
+### Declare your Api doc in files
+
+You can declare your api doc in YAML files if you enable the documentation_files in your config.yml
+ 
+ ```yaml
+ # app/config/config.yml
+ nelmio_api_doc:
+     documentation_files:  ~
```
Comment (OskarStark) [inline Resources/doc/index.md]: you should make an info for the default path

**Comment by OskarStark** [comment]
Status: reviewd

**Comment by willdurand** [comment]
Closing as I am not able to accept new features right now.

**Pull request closed without merging**